src/approval/simple_multisig.rs

```diff
@@ -0,0 +1,240 @@
+use std::{borrow::Cow, marker::PhantomData};
+
+use near_sdk::{
+    borsh::{self, BorshDeserialize, BorshSerialize},
+    env, require, AccountId,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::approval::{Approval, ApprovalState};
+
+pub trait SimpleMultisig<A: super::Action, P: SimpleMultisigApprover> {
+    fn root() -> crate::slot::Slot<()>;
+    fn init(config: SimpleMultisigConfig<P>)
+    where
+        Self: Sized,
+    {
+        <Self as Approval<A, SimpleMultisigApprovalState, SimpleMultisigConfig<P>>>::init(config);
+    }
+}
+
+impl<A, P, T> Approval<A, SimpleMultisigApprovalState, SimpleMultisigConfig<P>> for T
+where
+    A: super::Action,
+    P: SimpleMultisigApprover,
+    T: SimpleMultisig<A, P>,
+{
+    fn root() -> crate::slot::Slot<()> {
+        T::root()
+    }
+}
+
+pub trait SimpleMultisigApprover {
+    fn approve(account_id: &AccountId) -> Result<(), Cow<str>>;
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Debug)]
+pub struct SimpleMultisigConfig<A: SimpleMultisigApprover> {
+    pub threshold: u8,
+    #[borsh_skip]
+    #[serde(skip)]
```

Comment:
what is purpose of skipping serialization/deserialization here?

Reply:
`PhantomData` is a zero-sized type so it shouldn't matter in theory, but I think it's best-practice to avoid any nasty bugs later.